src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1527,6 +1528,53 @@ def reoffset(o):
 
         minimize_int(offset, reoffset)
 
+    def shrink_offset_pairs(self):
+        """Lower any two blocks offset from each other the same ammount.
+
+        Try lowering every non-zero pair of non-shrinking blocks by a
+        common offset.
+        """
+        self.debug('Shrinking offset pairs.')
+
+        def int_from_block(i):
+            u, v = self.blocks[i]
+            return int_from_bytes(self.shrink_target.buffer[u:v])
+
+        def block_len(i):
+            u, v = self.blocks[i]
+            return v - u
+
+        # Try reoffseting every pair
+        def reoffset_pair(pair, o):
+            n = len(self.blocks)
+            # Number of blocks may have changed, need to validate
+            valid_pair = [p for p in pair if p < n]
+
+            m = min([int_from_block(p) for p in valid_pair])
+
+            new_blocks = [self.shrink_target.buffer[u:v]
+                          for u, v in self.blocks]
+            for i in valid_pair:
+                if int_from_block(i) + o - m >= 0:
+                    new_blocks[i] = int_to_bytes(
+                        int_from_block(i) + o - m, block_len(i))
+            buffer = hbytes().join(new_blocks)
+            if sort_key(buffer) < sort_key(self.shrink_target.buffer):
```

Comment:
This condition is likewise always true and can simply be deleted - if `i` and `j` are always greater than zero, the constructed buffer will always be an improvement in terms of `sort_key`.  

(see next comment on maintaining that invariant)

As a design point, some of this code might be easier to read if the signature was `reoffset_pair(i, j)` and the offset calculated internally.

Reply:
Not sure I follow re: your last point. Isn't the point of a `minimize_int` call to calculate the best offset for an `(i,j)` pair? Not sure what calculating internally would look like.